src/test/frontend/replicasetlist/replicasetlist_controller_test.js

```diff
@@ -0,0 +1,45 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ReplicaSetListController from 'replicasetlist/replicasetlist_controller';
+import replicaSetListModule from 'replicasetlist/replicasetlist_module';
+
+describe('Replica set list controller', () => {
+  /**
+   * @type {!ReplicaSetCardController}
+   */
+  let ctrl;
+  /** @type {!ui.router.$state} */
+  let state;
+
+  beforeEach(() => {
+    angular.mock.module(replicaSetListModule.name);
+
+    angular.mock.inject(($controller, $state) => {
+      state = $state;
+      ctrl = $controller(ReplicaSetListController, {replicaSets: {replicaSets: []}});
+    });
+  });
+
+  it('should go to deploy page', () => {
```

Comment:
This is not integration test. It is only controller logic test. It checks whether the state of `ui.router.$state` object has changed to what is the expected value. This description fits better to integration tests where we can actually get button object and click it, then wait for browser response and check.

Description of the test may be misleading and I propose to change it to `should change state to deploy view on call`